sar/comm.py

```diff
@@ -196,9 +201,22 @@ def initialize_comms(_rank: int, _world_size: int, master_ip_address: str,
     os.environ['I_MPI_COMM_RANK'] = str(_rank)
 
     print("init_process_group: ", backend, _rank, _world_size)
+    
```

Comment:
Hey @Kacper-Pietkun this looks good. One thing I wasn't thinking about and didn't mention before (my apologies) is that this initialization should be backward compatible with the normal way of initializing comms in SAR, e.g., 

```
dist.init_process_group(
                backend=backend, rank=_rank, world_size=_world_size)
```

So it would be good to do a check that either *both* `shared_file` and `barrier` are provided or *neither* are. If one or the other is provided that would like be a mistake by the user. In normal operation neither would be provided and only in single-node mode should both be provided. 

After that check there should be a condition statement that checks if the initialization is for normal operation or single-node, and if single-node remove the shared file and call `dist.init_process_group` with `init_method=shared_file` otherwise just call `dist.init_process_group` as the above without passing `init_method`.